lib/ansible/module_utils/basic.py

```diff
@@ -2027,6 +2028,37 @@ def fail_json(self, **kwargs):
         self._return_formatted(kwargs)
         sys.exit(1)
 
+
+    def fail_json_aws(self, exception, msg=None):
+        """call fail_json with processed exception
+
+        function for converting exceptions thrown by AWS SDK modules,
+        botocore, boto3 and boto, into nice error messages.
+        """
+        last_traceback=traceback.format_exc()
+
+        try:
+            except_msg = exception.message
+        except AttributeError:
+            except_msg = str(exception)
+
+        if msg is not None:
+            message = '{0}: {1}'.format(msg, except_msg)
+        else:
+            message = except_msg
+
+        try:
+            response=exception.response
+        except AttributeError:
+            response=None
+
+        if response is None:
+            self.fail_json(msg=message, traceback=last_traceback)
```

Comment:
very funny.. I'm testing for the wrong thing so I write completely the wrong thing